scripts/build_dists_archive.sh

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env bash
+#
+# build zip for tagged releases
+#
+
+#shopt -s extglob
+
+# ToDo
+#  Check options
+#
+
+if [ "$1" = "-h" ] || [ "$1" = "--help" ]; then
+  echo "$0 (latest-tag|latest-tagv|'any-string-version')"
+  echo " 'latest-tag' pull and switch to latest git tag"
+  echo " 'latest-tagv' pull and switch to latest git tag starting with 'v'"
+  echo " 'any-string-version' archive string to tag with"
+  echo "   ie: $0 nightly-development-test-\$(date +'%Y-%m-%d')"
+  exit 1
+fi
+
+# Env
+distName="tari_base_node"
+sName=$(basename $0)
+#sPath=$(realpath $0)
+sPath=$(dirname $0)
+tsstamp=$(date +'%Y%m%dT%Hh%M')
+
+envFile="$sPath/.env"
+if [ -f "$envFile" ]; then
+  echo "Overriding Environment with $envFile file for settings ..."
+  source "$envFile"
+fi
+
+if [ -f "Cargo.toml" ]; then
+    echo "Cleaning Cargo ... "
+    cargo clean
```

Comment:
I do not think this (`cargo clean`) should be the default, or it should be able to bypass it with a command-line argument.